claim-check/code-samples/sample-4/client-producer/Program.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Configuration;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace client_producer
+{
+    class Program
+    {
+        public static PayloadDetails payloadDetails = new PayloadDetails();
+
+        public static void Main()
+        {
+            Console.WriteLine("Begin sample 4 for Claim n Check pattern");
+            Console.WriteLine();
+
+            Console.WriteLine("Creating payload and uploading it to Azure Blob Storage...");
+            ProcessAsync().GetAwaiter().GetResult();
+            Console.WriteLine("Done.");
+            Console.WriteLine();
+
+            Console.WriteLine("Sending Claim Check message...");
+            SendNotification();
+            Console.WriteLine("Done.");
+            Console.WriteLine();
+
+            Console.WriteLine("Press any key to exit the sample application.");
+            Console.ReadLine();
+        }
+
+        private static void SendNotification()
+        {
+            string brokerList = ConfigurationManager.AppSettings["EH_FQDN"];
+            string connectionString = ConfigurationManager.AppSettings["EH_CONNECTION_STRING"];
+            string topic = ConfigurationManager.AppSettings["EH_NAME"];
+            string caCertLocation = ConfigurationManager.AppSettings["CA_CERT_LOCATION"];
+
+            Worker.Producer(brokerList, connectionString, topic, caCertLocation, payloadDetails).Wait();
+        }
+
+        private static async Task ProcessAsync()
+        {
+            // Initialize variables
+            CloudStorageAccount storageAccount = null;
+            CloudBlobContainer cloudBlobContainer = null;
+            string sourceFile = null;
+
+            // Store message in blob storage
+            string storageConnectionString = ConfigurationManager.AppSettings["STORAGE_CONNECTION_STRING"];
+            
+            // Check whether the connection string can be parsed.
+            if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
+            {
+                try
+                {
+                    // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+                    CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+
+                    // Create a container called 'heavypayload' and append a GUID value to it to make the name unique. 
+                    var containerName = "heavypayload" + Guid.NewGuid().ToString();
+                    Console.WriteLine("- Creating container '{0}'", containerName);
+                    cloudBlobContainer = cloudBlobClient.GetContainerReference(containerName);
+                    await cloudBlobContainer.CreateAsync();
+
+                    // Set the permissions so the blobs are public. 
```

Comment:
We should not make the blobs public.  The connection string for the Azure function should have credentials that will allow it to read the blobs.

Reply:
The purpose of the sample is showing the claim-check pattern in the simplest way possible so that people can focus on it. All the other best practices can be found in their respective documentation. Using a SAS key will make the example more complex for no specific reason (related to the claim-check sample itself)

Reply:
I wasn't suggesting SAS.  You can specify the account name and key in the connection string and it will give access to the containers.  The user can copy it out of the Azure portal to put in config.  The Azure function can use the same connection string.

Reply:
Also, I would probably change the cloudBlobContainer initialization line to
```csharp
var cloudBlobContainer = cloudBlobClient.GetContainerReference(containerName);
```
and remove the variable declaration.

You can also change
```csharp
if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
```
to
```csharp
if (CloudStorageAccount.TryParse(storageConnectionString, out var storageAccount))
```
and remove it's declaration as well.

Reply:
Corrected